bowappend.go

```diff
@@ -14,42 +19,90 @@ func AppendBows(bows ...Bow) (Bow, error) {
 		return bows[0], nil
 	}
 
-	refBow := bows[0]
 	numRows := 0
 	for _, b := range bows {
-		if !b.Schema().Equal(refBow.Schema()) {
-			return nil,
-				fmt.Errorf("bow.AppendBow: schema mismatch: got both\n%v\nand\n%v",
-					refBow.Schema(), b.Schema())
-		}
-
-		if b.Metadata().String() != refBow.Metadata().String() {
-			return nil,
-				fmt.Errorf("bow.AppendBow: schema Metadata mismatch: got both\n%v\nand\n%v",
-					refBow.Metadata(), b.Metadata())
-		}
-
 		numRows += b.NumRows()
 	}
 
+	refBow := bows[0]
 	seriesSlice := make([]Series, refBow.NumCols())
-	bufSlice := make([]Buffer, refBow.NumCols())
+
 	for colIndex := 0; colIndex < refBow.NumCols(); colIndex++ {
-		var rowOffset int
+		var newArray array.Interface
+		mem := memory.NewCheckedAllocator(memory.NewGoAllocator())
 		typ := refBow.ColumnType(colIndex)
-		name := refBow.ColumnName(colIndex)
-		bufSlice[colIndex] = NewBuffer(numRows, typ, true)
-		for _, b := range bows {
-			for ri := 0; ri < b.NumRows(); ri++ {
-				bufSlice[colIndex].SetOrDrop(ri+rowOffset, b.GetValue(colIndex, ri))
+		switch typ {
```

Comment:
I think there's a lot of duplication in each case. I don't know if we can avoid that easily though